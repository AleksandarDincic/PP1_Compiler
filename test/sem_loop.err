ERROR 17:00:19,690 - Koriscenje break je dozvoljeno samo unutar petlji.
ERROR 17:00:19,690 - Koriscenje continue je dozvoljeno samo unutar petlji.
ERROR 17:00:19,690 - Koriscenje break je dozvoljeno samo unutar petlji.
ERROR 17:00:19,690 - Koriscenje continue je dozvoljeno samo unutar petlji.
ERROR 17:00:19,690 - Semanticka analiza neuspesno zavrsena.
