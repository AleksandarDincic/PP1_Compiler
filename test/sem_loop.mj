program sem_loop

{
	void main()
	int i;
	{
		do
		{
			break;
			continue;
		
			do
			{
				if(i < 0)
				{
				break;
				continue;
				}
			
				break;
				continue;
			
			}while(i < 0);
		}while(i < 0);
		
		if(i < 0)
		{
			break;	
			continue;
		}
		
		break;
		continue;
	}
}