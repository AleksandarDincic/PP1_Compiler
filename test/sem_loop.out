INFO  17:00:19,545 - Kompajliranje fajla: D:\Faks\PP1\workspace\MJCompiler\test\sem_loop.mj
INFO  17:00:19,546 - ===================================
INFO  17:00:19,546 - Leksicka analiza
INFO  17:00:19,548 - ===================================
INFO  17:00:19,548 - Sintaksna analiza
INFO  17:00:19,598 - #2 program
INFO  17:00:19,598 - #51 sem_loop
INFO  17:00:19,598 - #47 {
INFO  17:00:19,599 - #15 void
INFO  17:00:19,600 - #51 main
INFO  17:00:19,600 - #43 (
INFO  17:00:19,600 - #44 )
INFO  17:00:19,601 - #51 int
INFO  17:00:19,602 - #51 i
INFO  17:00:19,603 - #39 ;
INFO  17:00:19,603 - #47 {
INFO  17:00:19,603 - #9 do
INFO  17:00:19,604 - #47 {
INFO  17:00:19,604 - #3 break
INFO  17:00:19,604 - #39 ;
INFO  17:00:19,606 - #17 continue
INFO  17:00:19,607 - #39 ;
INFO  17:00:19,607 - #9 do
INFO  17:00:19,607 - #47 {
INFO  17:00:19,607 - #8 if
INFO  17:00:19,607 - #43 (
INFO  17:00:19,608 - #51 i
INFO  17:00:19,608 - #32 <
INFO  17:00:19,609 - #50 0
INFO  17:00:19,609 - #44 )
INFO  17:00:19,610 - #47 {
INFO  17:00:19,610 - #3 break
INFO  17:00:19,610 - #39 ;
INFO  17:00:19,610 - #17 continue
INFO  17:00:19,610 - #39 ;
INFO  17:00:19,610 - #48 }
INFO  17:00:19,610 - #3 break
INFO  17:00:19,611 - #39 ;
INFO  17:00:19,611 - #17 continue
INFO  17:00:19,611 - #39 ;
INFO  17:00:19,611 - #48 }
INFO  17:00:19,611 - #10 while
INFO  17:00:19,611 - #43 (
INFO  17:00:19,611 - #51 i
INFO  17:00:19,612 - #32 <
INFO  17:00:19,612 - #50 0
INFO  17:00:19,612 - #44 )
INFO  17:00:19,612 - #39 ;
INFO  17:00:19,612 - #48 }
INFO  17:00:19,613 - #10 while
INFO  17:00:19,613 - #43 (
INFO  17:00:19,613 - #51 i
INFO  17:00:19,613 - #32 <
INFO  17:00:19,613 - #50 0
INFO  17:00:19,613 - #44 )
INFO  17:00:19,613 - #39 ;
INFO  17:00:19,613 - #8 if
INFO  17:00:19,613 - #43 (
INFO  17:00:19,614 - #51 i
INFO  17:00:19,614 - #32 <
INFO  17:00:19,614 - #50 0
INFO  17:00:19,614 - #44 )
INFO  17:00:19,614 - #47 {
INFO  17:00:19,614 - #3 break
INFO  17:00:19,614 - #39 ;
INFO  17:00:19,614 - #17 continue
INFO  17:00:19,614 - #39 ;
INFO  17:00:19,614 - #48 }
INFO  17:00:19,614 - #3 break
INFO  17:00:19,614 - #39 ;
INFO  17:00:19,615 - #17 continue
INFO  17:00:19,615 - #39 ;
INFO  17:00:19,615 - #48 }
INFO  17:00:19,615 - #48 }
INFO  17:00:19,618 - Parsiranje uspesno zavrseno.
INFO  17:00:19,620 - Program(
  ProgramName(
   sem_loop
  ) [ProgramName]
  NoProgDecl(
  ) [NoProgDecl]
  MethodDeclarations(
    NoMethodDecl(
    ) [NoMethodDecl]
    MethodDecl(
      MethodReturnsVoid(
      ) [MethodReturnsVoid]
      MethodSignature(
       main
        NoFormalParams(
        ) [NoFormalParams]
      ) [MethodSignature]
      VarDeclarations(
        NoVarDecl(
        ) [NoVarDecl]
        VarDecl(
          VarType(
            Type(
             int
            ) [Type]
          ) [VarType]
          SingleVarName(
            VarNameOfSingle(
             i
            ) [VarNameOfSingle]
          ) [SingleVarName]
        ) [VarDecl]
      ) [VarDeclarations]
      ManyStatements(
        ManyStatements(
          ManyStatements(
            ManyStatements(
              NoStatement(
              ) [NoStatement]
              OneStatement(
                DoWhileStatement(
                  DoWhileLoopStart(
                  ) [DoWhileLoopStart]
                  BlockStatement(
                    Statements(
                      ManyStatements(
                        ManyStatements(
                          ManyStatements(
                            NoStatement(
                            ) [NoStatement]
                            OneStatement(
                              BreakStatement(
                              ) [BreakStatement]
                            ) [OneStatement]
                          ) [ManyStatements]
                          OneStatement(
                            ContinueStatement(
                            ) [ContinueStatement]
                          ) [OneStatement]
                        ) [ManyStatements]
                        OneStatement(
                          DoWhileStatement(
                            DoWhileLoopStart(
                            ) [DoWhileLoopStart]
                            BlockStatement(
                              Statements(
                                ManyStatements(
                                  ManyStatements(
                                    ManyStatements(
                                      NoStatement(
                                      ) [NoStatement]
                                      OneStatement(
                                        IfStatement(
                                          IfStart(
                                          ) [IfStart]
                                          IfCondition(
                                            ConditionTerm(
                                              CondTermFact(
                                                RelopCondFact(
                                                  PositiveExpr(
                                                    ExprTerm(
                                                      TermFactor(
                                                        DesignOfVarFactor(
                                                          NamedDesignator(
                                                           i
                                                          ) [NamedDesignator]
                                                        ) [DesignOfVarFactor]
                                                      ) [TermFactor]
                                                    ) [ExprTerm]
                                                  ) [PositiveExpr]
                                                  RelopLt(
                                                  ) [RelopLt]
                                                  PositiveExpr(
                                                    ExprTerm(
                                                      TermFactor(
                                                        ConstFactor(
                                                          IntConst(
                                                           0
                                                          ) [IntConst]
                                                        ) [ConstFactor]
                                                      ) [TermFactor]
                                                    ) [ExprTerm]
                                                  ) [PositiveExpr]
                                                ) [RelopCondFact]
                                              ) [CondTermFact]
                                            ) [ConditionTerm]
                                          ) [IfCondition]
                                          BlockStatement(
                                            Statements(
                                              ManyStatements(
                                                ManyStatements(
                                                  NoStatement(
                                                  ) [NoStatement]
                                                  OneStatement(
                                                    BreakStatement(
                                                    ) [BreakStatement]
                                                  ) [OneStatement]
                                                ) [ManyStatements]
                                                OneStatement(
                                                  ContinueStatement(
                                                  ) [ContinueStatement]
                                                ) [OneStatement]
                                              ) [ManyStatements]
                                            ) [Statements]
                                          ) [BlockStatement]
                                        ) [IfStatement]
                                      ) [OneStatement]
                                    ) [ManyStatements]
                                    OneStatement(
                                      BreakStatement(
                                      ) [BreakStatement]
                                    ) [OneStatement]
                                  ) [ManyStatements]
                                  OneStatement(
                                    ContinueStatement(
                                    ) [ContinueStatement]
                                  ) [OneStatement]
                                ) [ManyStatements]
                              ) [Statements]
                            ) [BlockStatement]
                            DoWhileCondStart(
                            ) [DoWhileCondStart]
                            ConditionTerm(
                              CondTermFact(
                                RelopCondFact(
                                  PositiveExpr(
                                    ExprTerm(
                                      TermFactor(
                                        DesignOfVarFactor(
                                          NamedDesignator(
                                           i
                                          ) [NamedDesignator]
                                        ) [DesignOfVarFactor]
                                      ) [TermFactor]
                                    ) [ExprTerm]
                                  ) [PositiveExpr]
                                  RelopLt(
                                  ) [RelopLt]
                                  PositiveExpr(
                                    ExprTerm(
                                      TermFactor(
                                        ConstFactor(
                                          IntConst(
                                           0
                                          ) [IntConst]
                                        ) [ConstFactor]
                                      ) [TermFactor]
                                    ) [ExprTerm]
                                  ) [PositiveExpr]
                                ) [RelopCondFact]
                              ) [CondTermFact]
                            ) [ConditionTerm]
                          ) [DoWhileStatement]
                        ) [OneStatement]
                      ) [ManyStatements]
                    ) [Statements]
                  ) [BlockStatement]
                  DoWhileCondStart(
                  ) [DoWhileCondStart]
                  ConditionTerm(
                    CondTermFact(
                      RelopCondFact(
                        PositiveExpr(
                          ExprTerm(
                            TermFactor(
                              DesignOfVarFactor(
                                NamedDesignator(
                                 i
                                ) [NamedDesignator]
                              ) [DesignOfVarFactor]
                            ) [TermFactor]
                          ) [ExprTerm]
                        ) [PositiveExpr]
                        RelopLt(
                        ) [RelopLt]
                        PositiveExpr(
                          ExprTerm(
                            TermFactor(
                              ConstFactor(
                                IntConst(
                                 0
                                ) [IntConst]
                              ) [ConstFactor]
                            ) [TermFactor]
                          ) [ExprTerm]
                        ) [PositiveExpr]
                      ) [RelopCondFact]
                    ) [CondTermFact]
                  ) [ConditionTerm]
                ) [DoWhileStatement]
              ) [OneStatement]
            ) [ManyStatements]
            OneStatement(
              IfStatement(
                IfStart(
                ) [IfStart]
                IfCondition(
                  ConditionTerm(
                    CondTermFact(
                      RelopCondFact(
                        PositiveExpr(
                          ExprTerm(
                            TermFactor(
                              DesignOfVarFactor(
                                NamedDesignator(
                                 i
                                ) [NamedDesignator]
                              ) [DesignOfVarFactor]
                            ) [TermFactor]
                          ) [ExprTerm]
                        ) [PositiveExpr]
                        RelopLt(
                        ) [RelopLt]
                        PositiveExpr(
                          ExprTerm(
                            TermFactor(
                              ConstFactor(
                                IntConst(
                                 0
                                ) [IntConst]
                              ) [ConstFactor]
                            ) [TermFactor]
                          ) [ExprTerm]
                        ) [PositiveExpr]
                      ) [RelopCondFact]
                    ) [CondTermFact]
                  ) [ConditionTerm]
                ) [IfCondition]
                BlockStatement(
                  Statements(
                    ManyStatements(
                      ManyStatements(
                        NoStatement(
                        ) [NoStatement]
                        OneStatement(
                          BreakStatement(
                          ) [BreakStatement]
                        ) [OneStatement]
                      ) [ManyStatements]
                      OneStatement(
                        ContinueStatement(
                        ) [ContinueStatement]
                      ) [OneStatement]
                    ) [ManyStatements]
                  ) [Statements]
                ) [BlockStatement]
              ) [IfStatement]
            ) [OneStatement]
          ) [ManyStatements]
          OneStatement(
            BreakStatement(
            ) [BreakStatement]
          ) [OneStatement]
        ) [ManyStatements]
        OneStatement(
          ContinueStatement(
          ) [ContinueStatement]
        ) [OneStatement]
      ) [ManyStatements]
    ) [MethodDecl]
  ) [MethodDeclarations]
) [Program]
INFO  17:00:19,620 - ===================================
INFO  17:00:19,620 - Semanticka analiza:
INFO  17:00:19,689 - Detektovan pristup lokalnoj promenljivoj na liniji 14
INFO  17:00:19,689 - 	Var i: int, 0, 1 
INFO  17:00:19,689 - Detektovan pristup lokalnoj promenljivoj na liniji 23
INFO  17:00:19,689 - 	Var i: int, 0, 1 
INFO  17:00:19,689 - Detektovan pristup lokalnoj promenljivoj na liniji 24
INFO  17:00:19,689 - 	Var i: int, 0, 1 
INFO  17:00:19,689 - Detektovan pristup lokalnoj promenljivoj na liniji 26
INFO  17:00:19,689 - 	Var i: int, 0, 1 
